Skeleton detachable body parts

flrsh opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->

[FEATURE] [CLEAN]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

With this PR, skeletons can now detach their limbs and head (without dying!). This PR also slightly cleans up / refactors some surgery code.

**Head**
* Once detached, the skeleton's "eye" is centered on the head object while their controls are still bound to their body
    * This means decapitation does not kill skeletons
    * Skeletons will not talk out of their heads, unfortunately, but rather still their bodies
* Cloths can be used on the head object to clean it
 
**Limbs**
* Once detached, the skeleton's limbs are functionally boomerangs that reattach upon catching and deal more damage than non-skeleton limbs
* "Joint wax" can be applied to empty limb sockets by targeting the respective limb; skeletons receive the "spry" status buff upon application 
    * Skeletons spawn with a small jar of joint wax to incentivize this practice
    * With the spry status buff, users move slightly faster and incur a smaller slowdown when injured 

**To detach:** Traditional surgery *or* crowbar + wrench + crowbar targeting respective body part. 
**To reattach:** Hold part in active hand and target respective body part. 

The sfx are by Virva (though I kinda mangled them), the spry status icon is by Dim, and the joint wax sprite is by Flab! 

https://streamable.com/83ssyk

## Why's this needed? <!-- Describe why you think this should be added to the game. -->

I think this is a cool flavor thing that makes skeletons more interesting to play. This also provides a basic framework for some other skeleton things I want to add later. 

## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete it.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```changelog
(u)Flourish
(*)Skeletons can now detach their head and limbs by targeting the respective part and using a wrench, crowbar, and wrench again. Once their head is detached, the skeleton's "eye" remains centered on their head object. Once a limb is detached, it can be used as a boomerang that automatically re-attaches upon catching. 
(*)Skeletons now also spawn with a jar of "joint wax" which can be applied to empty joint sockets for a small status buff. Check out the <a href="https://github.com/goonstation/goonstation/pull/8761" target="_blank">PR page</a> to learn more. Special thanks to Virva, DimWhat, and Flaborized for the sfx and sprites that made this possible!
```

flappybatpal commented:
Won't this mean katanas, airlock, firelock suicides and the decap deagle are no longer fatal?

Exester509 commented:
> 

Katana decap will still leave you bleeding a whole lot, and good luck preventing the antag from stealing your head and throwing it into a grinder with a blind, bleeding body. As for the deagle, I believe it removes your head altogether, doesn't it?

Confusedxoz commented:
Will this make getting item arms easier?

flrsh commented:
> Won't this mean katanas, airlock, firelock suicides and the decap deagle are no longer fatal to skeletons?

Yes. If that's an issue, I can change it so that only wrench - crowbar - wrench decapitation is nonfatal, and that all other methods are still fatal. Let me know!

> Will this make getting item arms easier?

Only in the sense that there's an additional way to remove them for skeletons.

> In the video, it shows the "eye" of the skeleton initially being on the head but then snapping to the body after some time, whats the deal with that? 

The "eye" snaps to the body when the head is picked up and attached.

frawhst commented:
the decapitation stuff seems fine; with the katana it's a coup de grace, deagle is an admin weapon, and if someone is trying to surgery your head off as a combat thing then they can just toss your head out an airlock which will create. Issues for you'

frawhst commented:
tangentially related; could examining the head display whether or not the person is conscious? could be confusing talking to a skeleton head while their body is in crit or otherwise unconscious somewhere and suddenly not getting a response with no indicator why

ThatFiveGuys commented:
Talking specifically in the video, 15 seconds in without head in hand the camera is centered on the headless body.  Also the text should probably still come out of the head even if its detached, seems funnier when the main body gets dragged away or whatever.

flrsh commented:
> tangentially related; could examining the head display whether or not the person is conscious? 

Yup, added!

> Talking specifically in the video, 15 seconds in without head in hand the camera is centered on the headless body. 

Ah yeah you're right, that was a bug caused by swapping positions with Mrs. Muggles. Fixed!

>Also the text should probably still come out of the head even if its detached, seems funnier when the main body gets dragged away or whatever. I dont know if the blindness of the disattached head is needed? While I do get that it can be used as a way to disguise as being dead effectively, I think the blindness would make that strat inviable and that strat seems fun.

These are due to code limitations unfortunately! (Unless I wanted to completely refactor how some of these systems work.)

ZeWaka commented on _std/macros/ismob.dm:
```diff
@@ -20,6 +20,7 @@
 #define iscow(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/cow))
 #define ispug(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/pug))
 #define isfrog(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/amphibian))
+#define isskeleton(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/skeleton))
```
rip `IS_BONER` 😩

ZeWaka commented on code/procs/jobprocs.dm:
```diff
@@ -681,6 +681,11 @@
 					if (src.limbs.r_arm)
 						qdel(src.limbs.r_arm.remove(0))
 				boutput(src, "<b>Your singular arm makes you feel responsible for crimes you couldn't possibly have committed.</b>" )
+
```
```suggestion

	// Special mutantrace items
```

flrsh replied:
Such a dishonest way of upping your commit count!

ZeWaka replied:
you don't need to add me as a co-author when merging

ZeWaka commented on code/obj/item/tool/wrench.dm:
```diff
@@ -23,6 +23,10 @@
 		BLOCK_SETUP(BLOCK_ROD)
 
+	attack(mob/living/carbon/M as mob, mob/user as mob)
+		if (!wrench_surgery(M, user))
```
these should really be named like `surgery_wrench` but beyond scope

ZeWaka commented on code/datums/mutantraces.dm:
```diff
@@ -1104,18 +1104,49 @@ TYPEINFO(/datum/mutantrace)
+		..()
+
+	get_desc()
+		if (uses > 0)
+			. += " It looks like it has [uses] applications left."
```
what's wrong with it saying 0 left?

flrsh replied:
Fair.

ZeWaka commented on code/obj/item/organ_holder.dm:
```diff
@@ -767,8 +769,6 @@
 					else
 						return 0
 				var/obj/item/organ/head/newHead = I
-				if (src.donor.client)
-					src.donor.client.mob = new /mob/dead/observer(src.donor)
```
don't see where this got moved

flrsh replied:
It's actually unnecessary cuz it only matters for brains, and brains do this already.

ZeWaka commented on code/obj/item/organs/head.dm:
```diff
@@ -333,7 +336,10 @@
 				boutput(user, "<span class='alert'>[src.name] is already wearing [src.glasses.name]!</span>")
 			return
-
+		if (istype(W, /obj/item/cloth))
+			playsound(src, "sound/items/towel.ogg", 25, 1)
+			user.visible_message("<span class='notice'>[user] [pick("buffs", "shines", "cleans", "wipes", "polishes")] [src] with [W].</span>")
+			return
```
don't see how this is cleaning off blood stains and such

flrsh replied:
It was just for flavor but I can make it clean blood stains and such too.

ZeWaka replied:
i thought you meant by
> Cloths can be used on the head object to clean it
that it actually cleaned it lol

ZeWaka commented on code/datums/mutantraces.dm:
```diff
@@ -1104,18 +1104,49 @@ TYPEINFO(/datum/mutantrace)
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "wax"
+	w_class = W_CLASS_SMALL
+	var/uses = 10
```
is there any way planned to refill this

flrsh replied:
Not in this PR but maybe in the future, possibly.

Sovexe commented on code/obj/item/mob_parts.dm:
```diff
@@ -259,66 +259,35 @@ ABSTRACT_TYPE(/obj/item/parts)
 
-			attachee.limbs.l_leg = src
-			attachee.limbs.r_leg = src
+			playsound(attachee, "sound/effects/attach.ogg", 50, 1)
```
single quotes are proffered when referencing sound files that don't require string operations

flrsh replied:
I always get confused about which to use, thanks!
